libvast/src/index/ip_index.cpp

```diff
@@ -80,10 +80,12 @@ ip_index::lookup_impl(relational_operator op, data_view d) const {
               || op == relational_operator::not_in))
           return caf::make_error(ec::unsupported_operator, op);
         auto topk = x.length();
-        if (topk == 0)
-          return caf::make_error(ec::unspecified,
-                                 "invalid IP subnet length: ", topk);
         auto is_v4 = x.network().is_v4();
+        // If we're in a IPv4/0 subnet, then we can blindly return all IPv6
+        // addresses. If we're in an IPv6/0 subnet, then we can't have any
+        // results at all.
+        if (topk == 0)
+          return is_v4 ? ~v4_.coder().storage() : ids{offset(), false};
```

Comment:
This doesn't pass some trivial tests for me:
```
# Ground truth for number of IPv4 addresses
> cat  libvast_test/artifacts/logs/zeek/conn.log | zeek-cut id.orig_h | grep -v '::' | grep -v '-' | wc -l
8302
# But the index doesn't agree
> vast -q count --estimate 'id.orig_h in 0.0.0.0/0'
160
# Negation is not taken into account
> vast -q count --estimate 'id.orig_h !in 0.0.0.0/0'
160
```

I'm surprised that this index has any v4 specific logic beyond handling of `::ff:x:x`.
`x.length()` should have been incremented by 96 if the address was formatted as v4 way earlier.
If `topk` is 0 here we should return all non-null values or nothing depending on the operator. This should also not be handled by a dedicated branch.

Reply:
We resolved this last issue together; thanks for the help!